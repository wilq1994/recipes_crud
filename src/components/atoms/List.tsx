import React from "react";
import styled from "styled-components";

const Wrapper = styled.ul`
  list-style: none;
  padding: 0;
  margin: 0;
  li {
    padding: 0.5em 0;
    & + li {
      border-top: 1px solid #e0e0e0;
    }
  }
`;

type Props = {
  list: string[];
};

const List = ({ list }: Props) => {
  return (
    <Wrapper>
      {list.map((item, id) => (
        <li key={id}>{item}</li>
      ))}
    </Wrapper>
  );
};

export default List;
