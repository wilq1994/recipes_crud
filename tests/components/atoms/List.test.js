import React from "react";
import { shallow } from "enzyme";

import List from "components/atoms/List";

describe("List", () => {
  it("should render and match snapshot", () => {
    const list = ["a", "b", "c"];
    const component = shallow(<List list={list} />);

    expect(component.exists()).toBeTruthy();
    expect(component.find("li").length).toBe(list.length);
    expect(component).toMatchSnapshot();
  });
});
