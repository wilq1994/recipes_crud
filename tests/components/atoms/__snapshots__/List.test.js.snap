// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render and match snapshot 1`] = `
<styled.ul>
  <li
    key="0"
  >
    a
  </li>
  <li
    key="1"
  >
    b
  </li>
  <li
    key="2"
  >
    c
  </li>
</styled.ul>
`;
